nix/magic-folder-env.nix

```diff
@@ -0,0 +1,12 @@
+{ pkgs }:
+let
+  magic-folder-env = pkgs.python2.buildEnv.override {
```

Comment:
> I don't think you should need to build an environment just containing this package, rather than just installing the package.

Installing it where?

Reply:
I guess I mean as a dependency of gridsync. As I understand it (see [docs](https://nixos.org/manual/nixpkgs/stable/#python), both `python.buildEnv` and `python.withPackages` are designed to create a deriviation where you can invoke python and access the specified packages; they incidentally also populate the path of that derivation with the scripts from those packages. However, the `magic-folder` derivation itself  already has the scripts in its path, which is all that gridsync uses.

Thus, using `python.buildEnv` just adds a layer of derivations that doesn't add anything we need. Though, as I said in my first comment, you may need `toPythonApplication` to avoid mixing python dependencies between the two.

Reply:
I haven't tried to understand what buildEnv or toPythonApplication do in much detail.  Empirically though, toPythonApplication doesn't seem to provide enough isolation:

```
these derivations will be built:                                                                                     
  /nix/store/qqdv0fv20bvn43ap9psn6pki7n3z8kd8-gridsync-v0.4.2.dev.drv                                                          
building '/nix/store/qqdv0fv20bvn43ap9psn6pki7n3z8kd8-gridsync-v0.4.2.dev.drv'...                                             
Sourcing python-remove-tests-dir-hook                                                                                  
Sourcing python-catch-conflicts-hook.sh                                                                                       
Sourcing python-remove-bin-bytecode-hook.sh                                                                            
Sourcing setuptools-build-hook                                                                                                             
Using setuptoolsBuildPhase                                                                                                         
Using setuptoolsShellHook                                                                                                              
Sourcing pip-install-hook                                                                                            
Using pipInstallPhase                                                                                                         
Sourcing python-imports-check-hook.sh                                                                                
Using pythonImportsCheckPhase                                                                                              
Sourcing python-namespaces-hook                                                                                        
Sourcing setuptools-check-hook                                                                                         
qtPreHook                                                                                                                              
unpacking sources                                                                                                                 
unpacking source archive /nix/store/74kz6wrfmixkx5w6429w0xk424vwh8yk-gridsync                                              
source root is gridsync                                                                                                  
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file gridsync/versioneer.pyc                                           
patching sources                                                                                                         
applying patch /nix/store/4vmaaqr9sga1nvhznwblywknwmzf04av-test_network.patch                                            
patching file tests/test_network.py                                                                                                    
configuring                                                                                                                                
no configure script, doing nothing                                                                                         
building                                                                                                               
Executing setuptoolsBuildPhase                                                                                             
Error in sitecustomize; set PYTHONVERBOSE for traceback:                                                                       
ImportError: This package should not be accessible on Python 3. Either you are trying to run from the python-future src folder or your installation of python-future is corrupted.
Error processing line 1 of /nix/store/lqxqvqyammbvmdz0k6ml5rrzabjl8jq6-python-2.7.17-env/lib/python2.7/site-packages/zope.event-4.4-py2.7-nspkg.pth:
                                                                                                                               
Fatal Python error: init_import_size: Failed to import the site module                                                                     
Python runtime state: initialized                                                                                      
Traceback (most recent call last):                                                                                                 
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/site.py", line 169, in addpackage                  
    exec(line)                                                                                                         
  File "<string>", line 1, in <module>                                                                                    
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/importlib/util.py", line 14, in <module>   
    from contextlib import contextmanager                                                                                          
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/contextlib.py", line 5, in <module> 
    from collections import deque                                                                                        
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/collections/__init__.py", line 27, in <module>
    from reprlib import recursive_repr as _recursive_repr                                                            
  File "/nix/store/kc431vsmyzwyp3ark674mpv55yjk8qqq-python2.7-future-0.18.2/lib/python2.7/site-packages/reprlib/__init__.py", line 7, in <module>
    raise ImportError('This package should not be accessible on Python 3. '                                                  
ImportError: This package should not be accessible on Python 3. Either you are trying to run from the python-future src folder or your installation of python-future is corrupted.
                                                                                                                           
During handling of the above exception, another exception occurred:                                                    
                                                                                                                           
Traceback (most recent call last):                                                                                       
  File "/nix/store/lqxqvqyammbvmdz0k6ml5rrzabjl8jq6-python-2.7.17-env/lib/python2.7/site-packages/site.py", line 73, in <module>
    __boot()                                                                                                                 
  File "/nix/store/lqxqvqyammbvmdz0k6ml5rrzabjl8jq6-python-2.7.17-env/lib/python2.7/site-packages/site.py", line 47, in __boot
    addsitedir(item)                                                                                       
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/site.py", line 208, in addsitedir
    addpackage(sitedir, name, known_paths) 
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/site.py", line 179, in addpackage
    import traceback                                   
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/traceback.py", line 3, in <module>
    import collections                                               
  File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/collections/__init__.py", line 27, in <module>
    from reprlib import recursive_repr as _recursive_repr
  File "/nix/store/kc431vsmyzwyp3ark674mpv55yjk8qqq-python2.7-future-0.18.2/lib/python2.7/site-packages/reprlib/__init__.py", line 7, in <module>
    raise ImportError('This package should not be accessible on Python 3. '
ImportError: This package should not be accessible on Python 3. Either you are trying to run from the python-future src folder or your installation of python-future is corrupted.
builder for '/nix/store/qqdv0fv20bvn43ap9psn6pki7n3z8kd8-gridsync-v0.4.2.dev.drv' failed with exit code 1
error: build of '/nix/store/qqdv0fv20bvn43ap9psn6pki7n3z8kd8-gridsync-v0.4.2.dev.drv' failed
```

Reply:
Hmm. I guess the nix documentation is quite inaccurate there (though there are quite a number of related bugs in the tracker.

This is probably fine as-is then. Though some experimenting gave me:
```nix
{ callPackage, buildEnv }:
let
  magic-folder = callPackage ./magic-folder.nix { };
  magic-folder-env = buildEnv {
    name = "magic-folder";
    paths = [ magic-folder ];
    pathsToLink = [ "/bin" ];
  };
in
  magic-folder-env
```
(though I''d be tempted to roll this into `magic-folder.nix`)